packages/pdf/src/RPA/PDF/keywords/document.py

```diff
@@ -0,0 +1,510 @@
+        target_pdf: str = None,
+        user_pwd: str = "",
+        owner_pwd: str = None,
+        use_128bit: bool = True,
+    ) -> None:
+        """Encrypt PDF document.
+
+        :param source_pdf: filepath to the source pdf
+        :param target_pdf: filepath to the target pdf, stored by default
+            to `output_directory`
+        :param user_pwd: allows opening and reading PDF with restrictions
+        :param owner_pwd: allows opening PDF without any restrictions, by
+            default same `user_pwd`
+        :param use_128bit: whether to 128bit encryption, when false 40bit
+            encryption is used, default True
+        """
+        self.switch_to_pdf_document(source_pdf)
+        reader = PyPDF2.PdfFileReader(self.ctx.active_fileobject)
+
+        default_output = Path(self.output_directory / "encrypted.pdf")
+        output_filepath = Path(target_pdf) if target_pdf else default_output
+
+        if owner_pwd is None:
+            owner_pwd = user_pwd
+        writer = PyPDF2.PdfFileWriter()
+        writer.appendPagesFromReader(reader)
+        writer.encrypt(user_pwd, owner_pwd, use_128bit)
+        with open(str(output_filepath), "wb") as f:
+            writer.write(f)
+
+    @keyword
+    def pdf_decrypt(
+        self, source_pdf: str, target_pdf: str, password: str
+    ) -> bool:
+        """Decrypt PDF with password.
+
+        :param source_pdf: filepath to the source pdf
+        :param target_pdf: filepath to the decrypted pdf
+        :param password: password as a string
+        :return: True if decrypt was successful, else False or Exception
+        :raises ValueError: on decryption errors
+        """
+        self.switch_to_pdf_document(source_pdf)
+        reader = PyPDF2.PdfFileReader(self.ctx.active_fileobject)
+        try:
+            match_result = reader.decrypt(password)
+
+            if match_result == 0:
+                raise ValueError("PDF decrypt failed.")
+            elif match_result == 1:
+                self.logger.info("PDF was decrypted with user password.")
+            elif match_result == 2:
+                self.logger.info("PDF was decrypted with owner password.")
+            else:
+                return False
+
+            self.save_pdf(source=None, target=target_pdf, custom_reader=reader)
+            return True
+
+        except NotImplementedError as e:
+            raise ValueError(
+                f"Document {source_pdf} uses an unsupported encryption method."
+            ) from e
+        except KeyError:
+            self.logger.info("PDF is not encrypted")
+            return False
+        return False
+
+    @keyword
+    def replace_textbox_text(self, text: str, replace: str, source_pdf: str = None):
+        """Replace text content of a textbox with something else in the PDF.
+
+        :param text: this text will be replaced
+        :param replace: used to replace `text`
+        """
+        if source_pdf or self.rpa_pdf_document is None:
```

Comment:
The source document handling seems to be a bit inconsistent. Sometimes it's switched automatically to the active document and sometimes not.